lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```diff
@@ -145,21 +144,22 @@ public function exec($statement)
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId($name = null) : string
     {
         if ($name === null) {
-            return false;
+            return '0';
         }
 
-        $sql    = 'SELECT ' . $name . '.CURRVAL FROM DUAL';
-        $stmt   = $this->query($sql);
-        $result = $stmt->fetch(\PDO::FETCH_ASSOC);
-
-        if ($result === false || !isset($result['CURRVAL'])) {
-            throw new OCI8Exception("lastInsertId failed: Query was executed but no result was returned.");
+        try {
+            $stmt = $this->prepare('SELECT ' . $name . '.CURRVAL FROM DUAL');
+            $stmt->execute();
+        } catch (OCI8Exception $exception) {
+            // In case no sequence value is available yet, we get the error:
+            // ORA-08002: sequence $name.CURRVAL is not yet defined in this session
+            return '0';
```

Comment:
Would it make sense to also check the error code? Otherwise, the DBAL may be suppressing irrelevant exceptions.

Reply:
Not sure if we can do that in a reliable way without starting to hit false-positives. The API is a bit too squishy to rely on precise codes.